packages/edit-post/src/components/header/writing-menu/index.js

```diff
@@ -37,11 +76,21 @@ function WritingMenu() {
 				scope="core/edit-post"
 				name="fullscreenMode"
 				label={ __( 'Fullscreen mode' ) }
-				info={ __( 'Work without distraction' ) }
+				info={ __( 'Show and hide admin UI' ) }
 				messageActivated={ __( 'Fullscreen mode activated' ) }
 				messageDeactivated={ __( 'Fullscreen mode deactivated' ) }
 				shortcut={ displayShortcut.secondary( 'f' ) }
 			/>
+			<PreferenceToggleMenuItem
+				scope="core/edit-post"
+				name="distractionFree"
+				toggleHandler={ toggleDistractionFree }
```

Comment:
Because we need other stuff to happen (e.g. close sidebars) not only to toggle.

Reply:
But why is this different than the other preference toggles above? the sidebars should close when the preference is set anyway.

Reply:
Yes. I did not alter the rendering of everything distraction free should close so that they render conditionally on the preference. Instead I created this cleanup handler. To have:

- the top toolbar disappear
- the inserter close
- the list view close
- the general sidebar close
- the 1st block focused

... when this preference is toggled on I'd have to pepper with checks for the `isDistractionFree` in all these places (the last one would probably still need a special handler). 

I think the answer to:

> why is this different than the other preference toggles above?

is that distraction free has effects across the UI along the lines of availability not only behavior.